user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+
+```
+riem_measures_wrapper <- function(start_date, end_date, station){
+  riem_measures(station = station, start_date = start_date, end_date = end_date)  
+}
+```
+
+ Each parameter can have following Attributes
+
+- name
+- displayName
+- colSpan
+- dataType
+- showLabel
+- defaultValue
+- required
+
+#### name (required)
+
+`name` of the input parameter. 
+
+#### displayName (required)
+
+`displayName` attribute is used for Display Name on Data Source Picker Dialog and Import Data Dialog.
+
+#### colSpan (required)
+
+`colSpan` attribute holds width of the input parameter on UI. Please use 5 for now.
+
+#### dataType (required)
+
+`dataType` attribute holds a type of the input parameter. Supported types are:
+
+- text
+- select
+- number
+- boolean
+
+We do not support Date type parameter for now.
+
+##### text
+
+if you use `text`, it becomes input field that accepts characters.
+
+##### select
+
+This is useful when you want to create a static List of Values.
+For example, if you want to create a time range List of Values, you can create it by specifying `options` and `itemDataType` like below. `options` an array of selector options and each option needs to have `label` and `value` attributes.If your option value is text, `itemDataType` should be set as `text`. If your option value is number, `itemDataType` should be number. Date is not supported for `itemDataType` for now.
+
+```
+{
+  "name": "time_range",
+  "displayName": "Time Range",
+  "dataType": "select",
+  "itemDataType" : "text",
+  "options": [
+    {"label":"Last 5 Years", "value":"5y"},
+    {"label":"Last 1 Hour", "value":"1h"},
+    {"label":"Last 4 Hours", "value":"4h"},
+    {"label":"Last 1 Day", "value":"1d"},
+    {"label":"Last 7 Day", "value":"7d"},
+    {"label":"Last 30 Days", "value":"30d"},
+    {"label":"Last 90 Days", "value":"90d"},
+    {"label":"Last 12 Months", "value":"1y"},
+    {"label":"From 2004", "value":"all"}
+  ],
+  "defaultValue" : "5y"
+}
+```  
+##### number
+
+If you use `number`, it becomes input field that accepts numeric value.
+
+##### boolean
+
+If you use `boolean`, you can define a list of values with two values (i.e. true/false)
+
+#### defaultValue
+
+If the parameter needs default value, you can set it through `defaultValue` attribute.
+
+#### required
+
+If you want to make a parameter as mandatory parameter, set the `required` attribute as `true`. By setting this to `true`, you can force users to enter values for these parameters and if a user clicks `Get Data` button without filling these parameters, he/she gets following error on Import Dialog.
+
+![](import/images/plugin_input_field_validation.png)
+
+## Create plugin R script file.
+
+If your plugin simply passes input parameter values from UI to the R function defined in `function` attribute, you can stub the R script. (i.e. all you need is create a single line `library(riem)` file and save it as `plugin.r`)
```

Comment:
"you can stub the R script." This was confusing to me. I guess something like this would better understood.
"If your plugin simply passes input parameter values from UI to existing R function, which you specified as `function` attribute, you don't need to define a new function in the R script."